diff --git a/libxputty/xputty/xwidget.c b/libxputty/xputty/xwidget.c
index b4e3dac..a5ab5ac 100644
--- a/libxputty/xputty/xwidget.c
+++ b/libxputty/xputty/xwidget.c
@@ -101,8 +101,8 @@ void destroy_widget(Widget_t * w, Xputty *main) {
         cairo_destroy(w->cr);
         cairo_surface_destroy(w->surface);
         
-        XDestroyIC(w->xic);
-        XCloseIM(w->xim);
+        if (w->xic) XDestroyIC(w->xic);
+        if (w->xim) XCloseIM(w->xim);
         XUnmapWindow(w->app->dpy, w->widget);
         XDestroyWindow(w->app->dpy, w->widget);
         free(w->childlist);
@@ -167,17 +167,20 @@ Widget_t *create_window(Xputty *app, Window win,
                             CopyFromParent, &attributes);
     debug_print("XCreateWindow\n");
 
+#ifndef ANDROID
     XSetLocaleModifiers("");
     w->xim = XOpenIM(app->dpy, 0, 0, 0);
     if(!w->xim){
         XSetLocaleModifiers("@im=none");
         w->xim = XOpenIM(app->dpy, 0, 0, 0);
     }
-
     w->xic = XCreateIC(w->xim, XNInputStyle, XIMPreeditNothing | XIMStatusNothing,
                     XNClientWindow, w->widget, XNFocusWindow,  w->widget, NULL);
-
     XSetICFocus(w->xic);
+#else
+    w->xim = NULL;
+    w->xic = NULL;
+#endif
 
     XSelectInput(app->dpy, w->widget, event_mask);
 
@@ -297,17 +300,20 @@ Widget_t *create_widget(Xputty *app, Widget_t *parent,
                             CopyFromParent|CWOverrideRedirect, &attributes);
     debug_print("XCreateWindow\n");
 
+#ifndef ANDROID
     XSetLocaleModifiers("");
     w->xim = XOpenIM(app->dpy, 0, 0, 0);
     if(!w->xim){
         XSetLocaleModifiers("@im=none");
         w->xim = XOpenIM(app->dpy, 0, 0, 0);
     }
-
     w->xic = XCreateIC(w->xim, XNInputStyle, XIMPreeditNothing | XIMStatusNothing,
                     XNClientWindow, w->widget, XNFocusWindow,  w->widget, NULL);
-
     XSetICFocus(w->xic);
+#else
+    w->xim = NULL;
+    w->xic = NULL;
+#endif
 
     XSelectInput(app->dpy, w->widget, event_mask);
 
